src/mutations/appAction.ts

```diff
@@ -27,7 +27,7 @@ export default (queryConfig: QueryClientConfig) => {
           const newData: AppAction = newAppAction;
           // check that the websocket event has not already been received and therefore the data were added
           if (!prevData) {
-            queryClient.setQueryData(key, newData);
+            queryClient.setQueryData(key, [newData ?? []]);
```

Comment:
Was this the thing that was causing you issues ? But the backend is supposed to send you an array always no ? Really strange !

Reply:
Yes it seems to be the problem. The postAppAction return an AppAction and not an array in the backend so I don't really know what should be the behaviour of the setQueryData. Enforcing to save as an array solve the s.some is not a function.

Reply:
@ReidyT Ok, I got it. The issue is actually not in the backend. It is a small confusion here:
- POST returns only the posted appAction object
- GET returns an array of appActions
- When we get the data from the cache we will get an array of appAction
- When there is nothing in the cache, we should add our new data that we got from the POST call to the cache entry. But we have to add it as an array !
Your change is right !